### ISSUE TITLE
Traces into safespaces don't check for occluding entities

### ISSUE THREAD

**Issue by Fasteroid** [MattJeanes/Safe-Space#22]
![20210205154603_1](https://user-images.githubusercontent.com/29342750/107087673-cd934400-67c9-11eb-9b9e-9a27f3cf7d8e.jpg)
The cursed ranger beams are not your fault—I know those aren't fixable.  Ignore those for this bug report.

This screenshot shows both a prop and the world occluding rangers pointed at a safespace.  These rangers shouldn't enter the safespace since their beams should something else first—but they do.  It appears that so long as a trace is pointed at a safespace and is long enough, it will enter the safespace—regardless of whether or not something is between it and the safespace.

This happens on the workshop version.  Not sure about the github version since the last updates were 5 years ago.

**Comment by MattJeanes** [comment]
Yeah I thought it might do this... I'll move this issue to the base world-portals where the problem for this actually is
